packages/editor-ui/src/push-connection/__tests__/useEventSourceClient.test.ts

```diff
@@ -0,0 +1,173 @@
+import { afterEach, beforeEach, describe, expect, test, vi } from 'vitest';
+import { useEventSourceClient } from '../useEventSourceClient';
+
+/** Mocked EventSource class to help testing */
+class MockEventSource extends EventTarget {
```

Comment:
Would it make sense to extract this and `MockWebSocket` to `src/__tests__/mocks.ts`?

Reply:
Good point. Moved the mocks to their own files